cl/recap/mergers.py

```diff
@@ -1658,9 +1658,21 @@ async def merge_attachment_page_data(
                 .afirst()
             )
         else:
-            main_rd = await RECAPDocument.objects.select_related(
-                "docket_entry", "docket_entry__docket"
-            ).aget(**params)
+            try:
+                main_rd = await RECAPDocument.objects.select_related(
+                    "docket_entry", "docket_entry__docket"
+                ).aget(**params)
+            except RECAPDocument.DoesNotExist as exc:
+                # In cases where we have "doppelgänger" dockets drop pacer
+                # case id and check if the docket exists once more.
+                if params.get("pacer_case_id"):
```

Comment:
this `if` should have an `else` clause that re-raises the exception `exc`
```
if params.get("pacer_case_id"):
     ....
else:
    raise exc
```

[Tests](https://github.com/freelawproject/courtlistener/actions/runs/11712563431/job/32629242434?pr=4665) are pointing at an ``UnboundLocalError: cannot access local variable 'main_rd' where it is not associated with a value` caused by lack of that else clause

Reply:
Thanks @grossir for the second look